runtime/compiler/control/JITServerHelpers.cpp

```diff
@@ -37,63 +37,88 @@ bool         JITServerHelpers::_serverAvailable = true;
 uint64_t     JITServerHelpers::_nextConnectionRetryTime = 0;
 TR::Monitor *JITServerHelpers::_clientStreamMonitor = NULL;
 
+
+// To ensure that the length fields in UTF8 strings appended at the end of the
+// packed ROMClass are properly aligned, we must pad the strings accordingly.
+// This function returns the total size of a UTF8 string with the padding.
 static size_t
-methodStringsLength(J9ROMMethod *method)
+getUTF8Size(const J9UTF8 *str)
    {
-   J9UTF8 *name = J9ROMMETHOD_NAME(method);
-   J9UTF8 *sig = J9ROMMETHOD_SIGNATURE(method);
-   return (name->length + sig->length + (2 * sizeof(U_16)));
+   return OMR::alignNoCheck(str->length, sizeof(*str)) + sizeof(*str);
```

Comment:
Don't we add too much now? sizeof(*str) is 4 so we pad payload to be a multiple of 4 and then add another 4 bytes

Reply:
sizeof(*str) == sizeof(J9UTF8) == sizeof(J9UTF8::length) == 2. This function returns the size of the whole thing, including the header. The only change I made here was to use sizeof(J9UTF8) instead of sizeof(J9UTF8::length) because it exposes less details about J9UTF8 internals.

Reply:
Right. My IDE found me another definition in `genBinaryBlob.hpp`
```
	typedef struct J9UTF8 {
		uint16_t length;
		uint8_t data[2];
	} J9UTF8;
```
which may be used for DDR?